src/main/java/com/commercetools/project/sync/SyncerFactory.java

```diff
@@ -55,122 +61,172 @@ public static SyncerFactory of(
+    }
+    return CompletableFuture.allOf(syncersToRunParallel.toArray(new CompletableFuture[0]));
+  }
+
+  @Nonnull
+  private static List<SyncModuleOption> validateAndCollectSyncOptionValues(
+      @Nonnull final String[] syncOptionValues) {
 
-    if (isBlank(syncOptionValue)) {
+    if (null == syncOptionValues) {
       final String errorMessage =
           format(
-              "Blank argument supplied to \"-%s\" or \"--%s\" option! %s",
-              SYNC_MODULE_OPTION_SHORT, SYNC_MODULE_OPTION_LONG, SYNC_MODULE_OPTION_DESCRIPTION);
+              "Unknown argument \"%s\" supplied to \"-%s\" or \"--%s\" option! %s",
+              null,
+              SYNC_MODULE_OPTION_SHORT,
+              SYNC_MODULE_OPTION_LONG,
+              SYNC_MODULE_OPTION_DESCRIPTION);
+      throw new IllegalArgumentException(errorMessage);
+    }
+    if (syncOptionValues.length == 1 && SYNC_MODULE_OPTION_ALL.equals(syncOptionValues[0])) {
+      return Arrays.asList(SyncModuleOption.values());
+    } else {
+      return Arrays.stream(syncOptionValues)
+          .map(
+              syncOption -> {
+                if (isBlank(syncOption)) {
+                  final String errorMessage =
+                      format(
+                          "Blank argument supplied to \"-%s\" or \"--%s\" option! %s",
+                          SYNC_MODULE_OPTION_SHORT,
+                          SYNC_MODULE_OPTION_LONG,
+                          SYNC_MODULE_OPTION_DESCRIPTION);
 
-      return exceptionallyCompletedFuture(new IllegalArgumentException(errorMessage));
+                  throw new IllegalArgumentException(errorMessage);
+                }
+
+                final String trimmedValue = syncOption.trim();
+                try {
+                  return SyncModuleOption.getSyncModuleOptionBySyncOptionValue(trimmedValue);
+                } catch (IllegalArgumentException e) {
+                  final String errorMessage =
+                      format(
+                          "Unknown argument \"%s\" supplied to \"-%s\" or \"--%s\" option! %s",
+                          trimmedValue,
+                          SYNC_MODULE_OPTION_SHORT,
+                          SYNC_MODULE_OPTION_LONG,
+                          SYNC_MODULE_OPTION_DESCRIPTION);
+                  throw new IllegalArgumentException(errorMessage);
+                }
+              })
+          .collect(Collectors.toList());
     }
+  }
 
-    Syncer<
-            ? extends ResourceView,
-            ?,
-            ? extends BaseSyncStatistics,
-            ? extends BaseSyncOptions<?, ?>,
-            ? extends QueryDsl<?, ?>,
-            ? extends BaseSync<?, ?, ?>>
-        syncer;
+  private static Collection<List<SyncModuleOption>> groupSyncModuleOptions(
+      List<SyncModuleOption> syncModuleOptions) {
+    Map<SyncModuleOption, Integer> syncModuleOptionWithReferencesCount =
+        syncModuleOptions
```

Comment:
Did not understand this suggestion?

Reply:
You should have at most one stream method call per line. This will make stream operations like map, filter and collect easily recognizable. `stream()` is usually inline with the collection that you start to stream.

Reply:
Ya but It is formatted by spotlessApply. I could not change it to one line.

Reply:
ohh okay, It seems spotless does it in different. Whenever stream is passing as a param, it in-line the stream, but whenever it written as sole statement, it breaks. So, let's leave how spotless formatting it.